interp/constrintern.ml

```diff
@@ -761,23 +761,24 @@ let traverse_binder intern_pat ntnvars (terms,_,binders,_ as subst) avoid (renam
       let na = out_patvar pat in
       let env = push_name_env ntnvars [] env na in
       let ty' = DAst.make @@ GHole (Evar_kinds.BinderType na.CAst.v,IntroAnonymous,None) in
-      (renaming,env), None, na.v, bk, set_type ty (Some ty')
+      ((avoid,renaming),env), None, na.v, bk, set_type ty (Some ty')
     else
       (* Interpret as a pattern *)
       let env,((disjpat,ids),id),na,bk,t = intern_pat test_kind ntnvars env bk pat in
       let pat, na =
         match disjpat with
         | [pat] when is_patvar_store store pat -> let na = get () in None, na
         | _ -> Some ((List.map (fun x -> x.v) ids,disjpat),id), na in
-      (renaming,env), pat, na.v, bk, set_type ty (Some t)
+      ((avoid,renaming),env), pat, na.v, bk, set_type ty (Some t)
   with Not_found ->
     (* Binders not bound in the notation do not capture variables *)
     (* outside the notation (i.e. in the substitution) *)
     let id' = find_fresh_name renaming subst avoid id in
-    let renaming' =
+    let renaming =
       if Id.equal id id' then renaming else Id.Map.add id id' renaming
```

Comment:
Actually, `renaming` contributes to the names to avoid in `find_fresh_name`. I don't know well mangling but maybe an alternative would be to simply do `Id.Map.add id id' renaming` even when `id'` is the same as `id`.

Reply:
let's see what ci says